### ISSUE TITLE
Advance reporting is not working for Magento 2

### ISSUE THREAD

**Issue by Ridik123** [magento/magento2#21452]
I have followed all instruction given on Magento guideline.
URL: https://docs.magento.com/m2/ce/user_guide/reports/advanced-reporting.html
https://nimb.ws/5G31mD
https://nimb.ws/5XiUAr
the advance reporting status is changed from pending to enabled
https://nimb.ws/mXhLG2

I have reauthorized the analytic user
https://nimb.ws/0rkEPl

waited for 2-3 days but the advance reporting leads to 404 page
https://nimb.ws/KUCAB6

**Comment by magento-engcom-team** [comment]
Hi @Ridik123. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@Ridik123 do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by ghost** [comment]
HI @Ridik123  what the reason open new issue ?

**Comment by Ridik123** [comment]
as i am unable to subit ticket into magento??? how can i reach them

**Comment by ghost** [comment]
@Ridik123 i was leave a comment on previous ticket that this must be reported on forums, because on clean 2.3-dev on my environment everything work as expected

**Comment by Ridik123** [comment]
I have written there as well but getting no support

**Comment by ghost** [comment]
@Ridik123 but this is github issue tracker but not support :)

**Comment by Ridik123** [comment]
can't anyone help me in this issue? or guide me

**Comment by ghost** [comment]
@Ridik123 please verify that your site have this options

  | The website must run on a public web server.
-- | --
  | The domain must have a valid security (SSL) certificate.
  | Magento must have been installed or upgraded successfully without error.
  | In the Magento configuration, the Base URL (Secure) setting for the store view must point to the secure URL. For example https://yourdomain.com
  | In the Magento configuration, Use Secure URLs on Storefront, and Use Secure URLs in Admin must be set to “Yes.”
  | Make sure that Magento crontab is created and cron jobs are running on the installed server.

**Comment by Ridik123** [comment]
Hi have already rechecked these steps and all are good but still, the reporting is not working

**Comment by ghost** [comment]
@Ridik123 did you check if cron runs the job ?

**Comment by Ridik123** [comment]
yes crons are runnig fine as well

**Comment by ghost** [comment]
@Ridik123 wich your magento version ?

**Comment by Ridik123** [comment]
magento 2.2.5

**Comment by ghost** [comment]
@Ridik123 If you get a 404 “Page Not Found” message, verify that your store meets the requirements for Advanced Reporting. Then, follow the instructions to verify that the integration is installed.
you have some missing setiings -> https://github.com/magento/magento2/issues/21452#issuecomment-467346195

**Comment by Ridik123** [comment]
As I said the status for advanced reporting is changed from pending to enabled.

**Comment by Ridik123** [comment]
I have cross checked all the points twice

**Comment by ghost** [comment]
@Ridik123 contact your hosting provider,  issue must be in your environment

**Comment by Ridik123** [comment]
what king of issue would it be

as i have checked the logs and found that one the cron has this
setup-cron.ERROR: Your current PHP memory limit is 512M. Magento 2 requires it to be set to 756M or more. As a user with root privileges, edit your php.ini file to increase memory_limit. (The command php --ini tells you where it is located.) After that, restart your web server and try again.

**Comment by ghost** [comment]
@Ridik123 so you need to add more memory_limit `Increase the PHP memory limit to at least 768MB for normal operation or 2GB for testing.`

**Comment by Ridik123** [comment]
will that thing gonna resolve this

**Comment by Ridik123** [comment]
Increased limit now no-error in logs but still it leads to 404

**Comment by ghost** [comment]
@Ridik123 https://community.magento.com/t5/Magento-2-x-Technical-Issues/Problem-with-Advanced-Reporting/m-p/115099

**Comment by Ridik123** [comment]
ya I have checked this before posting this issue but the solution didn't work

**Comment by Ridik123** [comment]
Hello,

I am happy to inform you that there is a patch available to fix this issue in the core Magento installation, which I have attached here. Please upload the appropriate patch into your Magento root directory and run the SSH command below.

patch -p1 < MDVA-15136_EE_2.2.5_COMPOSER_v1.patch
or
patch -p1 < MDVA-15136_EE_2.2.5_v1.patch

Once that is done, refresh the cache in the Admin under "System > Cache Management" so that the changes will be reflected.

Please try it on a test environment first before taking it live, and kindly confirm that patch provided has resolved your issue.
[patches.zip](https://github.com/magento/magento2/files/2909595/patches.zip)

**Comment by wjpbill** [comment]
Did this patch work for anyone?
This problem has been driving me around the bend for months.
I've got a live 2.2.5 site and wondered if I should use the patch. I can't run it on Dev as that's 2.2.7

I'm also not sure on which version of the patch to run and what the difference is between them.

Would appreciate the feedback everyone

**Comment by Ridik123** [comment]
I am not sure about 2.2.7 but these patches work absolutely fine for me on 2.2.5.

**Comment by wjpbill** [comment]
> 
> 
> I am not sure about 2.2.7 but these patches work absolutely fine for me on 2.2.5.

Thank you for the reply @Ridik123 

Sorry to sound stupid but any ideas on what the differences between the two versions?
Is one just a composer install version?

**Comment by Ridik123** [comment]
@wjpbill  the major changes on these versions are based over security concerns.
I suggest you make a clone of the current live version and try on it.

**Comment by wjpbill** [comment]
> 
> 
> @wjpbill the major changes on these versions are based over security concerns.
> I suggest you make a clone of the current live version and try on it.

@Ridik123 Apologies, I don't think I was clear.

It says
patch -p1 < MDVA-15136_EE_2.2.5_COMPOSER_v1.patch
OR
patch -p1 < MDVA-15136_EE_2.2.5_v1.patch

is the first one, just a composer version of the same code do you know?

**Comment by gcampedelli** [comment]
Hello, what does Magento team says about the patch, should we apply? I have exactly same problem and I doubled checked all requirements as well, memory, php.ini and so on. I have exactly same configuration on another Magento install that surprisely works but no other does the same. Advanced Reports stopped working for us in Magento 2.2.6. Is the patch suitable for this version?

**Comment by Ridik123** [comment]
@wjpbill @gcampedelli these patches are provided from Magento support.
and I tried to install the first one which works for me so I didn't go for the second one

**Comment by Ridik123** [comment]
> > @wjpbill the major changes on these versions are based over security concerns.
> > I suggest you make a clone of the current live version and try on it.
> 
> @Ridik123 Apologies, I don't think I was clear.
> 
> It says
> patch -p1 < MDVA-15136_EE_2.2.5_COMPOSER_v1.patch
> OR
> patch -p1 < MDVA-15136_EE_2.2.5_v1.patch
> 
> is the first one, just a composer version of the same code do you know?

not sure but this one works for me so I didn't go for the second one

**Comment by bhargavmehta** [comment]
Advanced Reporting is not working even in Magento 2.3. It shows enabled in the backend but when we go to reporting link it shows 404.

Everything is configured as per the guide

**Comment by Ridik123** [comment]
I had a chat with Magento support and they said it is fixed in 2.3

**Comment by bhargavmehta** [comment]
Is it fixed in 2.3.0 or 2.3.1?

Thank you

**Comment by Ridik123** [comment]
@bhargavmehta I have not checked for these versions of Magento. But I guess the latest version which is released today have the fix for advance reporting

**Comment by gcampedelli** [comment]
@Ridik123 Thank you!

**Comment by hafizjee** [comment]
> I had a chat with Magento support and they said it is fixed in 2.3

it's not working in magento2.3 CE . 
Everything is configured as per the guide.
It shows enabled but when we go to reporting link it shows 404.

**Comment by adrian-martinez-interactiv4** [comment]
> > I had a chat with Magento support and they said it is fixed in 2.3
> 
> it's not working in magento2.3 CE .
> Everything is configured as per the guide.
> It shows enabled but when we go to reporting link it shows 404.

I was in the same situation, and I found out that it was a db data inconsistency issue.
More info here: https://community.magento.com/t5/Magento-2-x-Technical-Issues/Advanced-Reporting-doesn-t-work-and-there-is-an-error-404/m-p/145064/highlight/true#M11226

**Comment by davidduong90** [comment]
I used Magento 2.4.2 and I have the same issue.
